package_manager/parse_metadata_test.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+import os
+
+from parse_metadata import parse_package_metadata
+
+class TestParseMetadata(unittest.TestCase):
+
+    def setUp(self):
+        dir = os.path.dirname(__file__)
+        filename = os.path.join(dir, 'testdata', 'Packages.txt')
+        with open(filename) as f:
+            data = f.read()
+        self.data = data
+        self.mirror_url = "http://debian.org"
+        self.metadata = parse_package_metadata(self.data, self.mirror_url)
+
+    def test_url_rewrite(self):
+        self.assertEqual(
+            self.metadata["libnewlib-dev"]["Filename"],
+            self.mirror_url + "/debian/" + "pool/main/n/newlib/libnewlib-dev_2.1.0+git20140818.1a8323b-2_all.deb")
+
+    def test_get_all_packages(self):
+        expected_packages = ["libnewlib-dev",
+                             "libnewlib-doc",
+                             "newlib-source",
+                             "newmail",
+                             "zzuf",]
+        for pkg in expected_packages:
+            ok = self.metadata[pkg]
```

Comment:
What is this supposed to do, just check that the package is in metadata? You can use something like self.assertContains instead.